services/libs/aster-std/src/syscall/alarm.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::time::Duration;
+
+use super::{SyscallReturn, SYS_ALARM};
+use crate::log_syscall_entry;
+use crate::prelude::*;
+use crate::process::posix_thread::{PosixThreadExt, Timer};
+use crate::process::signal::constants::SIGALRM;
+use crate::process::signal::signals::kernel::KernelSignal;
+use crate::thread::thread_table;
+
+pub fn sys_alarm(seconds: u32) -> Result<SyscallReturn> {
+    log_syscall_entry!(SYS_ALARM);
+    debug!("seconds = {}", seconds);
+
+    let current_thread = current_thread!();
+    let mut timers = {
+        let posix_thread = current_thread.as_posix_thread().unwrap();
+        posix_thread.timers().lock()
```

Comment:
The alarm system call only manipulates one timer. This implementation is incorrect.